src/components/model-authoring/edit-name.tsx

```diff
@@ -0,0 +1,38 @@
+import * as React from "react";
+import { useState } from "react";
+
+import * as css from "./edit-name.scss";
+
+interface IProps {
+  name: string;
+  saveName: (name: string) => void;
+}
+
+export const EditName = ({ name, saveName }: IProps) => {
+  const [newName, setNewName] = useState<string>(name);
+  const [disabled, setDisabled] = useState<boolean>(true);
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setNewName(e.target.value);
+  }
+
+  const handleClick = () => {
+    setDisabled(false);
+  }
```

Comment:
If you rename the state updater this `setEditing` you can rename this `handleEditClick`.